app/bundles/UserBundle/Entity/UserInvite.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\UserBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Mautic\CoreBundle\Doctrine\Mapping\ClassMetadataBuilder;
+
+class UserInvite
+{
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $email;
+
+    /**
+     * @var string
+     */
+    private $token;
+
+    /**
+     * @var \DateTimeInterface
+     */
+    private $expiration;
+
+    /**
+     * @var bool
+     */
+    private $used = false;
+
+    /**
+     * @var Role|null
+     */
+    private $role;
```

Comment:
```suggestion
class UserInvite
{
    private ?int $id;

    private ?string $email;

    private ?string $token;

    private ?\DateTimeInterface $expiration;

    private bool $used = false;

    private ?Role $role;
```
please consider that if some property is required to create the UserInvite entity then create a constructor that will require such information. Like the $role for example. Then we can remove the nullable type and the code will get simpler as we will always know that a role is set and you don't have to check if it exists on multiple places.